inbox/models/when.py

```diff
@@ -65,40 +59,36 @@ def parse_keys(cls, raw):
                 raise ValueError(f"'{key}' parameter invalid.")
         return times
 
-    def __init__(self, start, end=None):
+    def __init__(self, start: arrow.Arrow, end: Optional[arrow.Arrow] = None):
         self.start = start
         self.end = end or start
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"{type(self)} ({self.start} - {self.end})"
 
     @property
-    def is_time(self):
+    def is_time(self) -> bool:
         return isinstance(self, Time)
 
     @property
-    def is_date(self):
+    def is_date(self) -> bool:
         return isinstance(self, Date)
 
     @property
-    def delta(self):
+    def delta(self) -> datetime.timedelta:
         return self.end - self.start
 
-    def get_time_dict(self):
+    def get_time_dict(self) -> Dict[str, arrow.Arrow]:
         times = (self.start, self.end)
         return dict(zip(self.json_keys, times))
 
 
-class AllDayWhen(When):
-    all_day = True
```

Comment:
This class was used just to set `all_day = True`, which is too little to have a separate class